controllers/topology/desired_state.go

```diff
@@ -229,9 +229,25 @@ func computeControlPlaneVersion(s *scope.Scope) (string, error) {
 		return "", errors.Wrap(err, "failed to get the version from control plane spec")
 	}
 
-	// Return early if the version is already equal to the desired version
-	// no further checks required.
+	s.UpgradeTracker.ControlPlane.PendingUpgrade = true
 	if *currentVersion == desiredVersion {
+		// Mark that the control plane spec is already at the desired version.
+		// This information is used to show the appropriate message for the TopologyReconciled
+		// condition.
+		s.UpgradeTracker.ControlPlane.PendingUpgrade = false
+	}
```

Comment:
Why are we tracking this here instead of looking at the current state of things when we need to calculate the condition?

Reply:
Since other information like `IsUpgrading`, `IsScaling` are being calculated here it made sense to keep all the code that builds information into the `UpgradeTracker` in one place instead of distributing the code that calculated the  TopologyReconciled condition.

Reply:
Infact thinking about it, we can reuse this `s.UpgradeTracker.ControlPlane` in `computeMachineDeploymentVersion`. This would make that function a lot simpler.

Reply:
I also like the effect of keeping the logic here in one place instead of distributing it.